File: crowdin/src/main/java/com/crowdin/platform/utils/FileUtils.java
package com.crowdin.platform.utils;

import android.content.Context;
import android.support.annotation.Nullable;

import com.crowdin.platform.api.ResourcesResponse;
import com.google.gson.Gson;

import java.io.IOException;
import java.io.InputStream;

public class FileUtils {

    private FileUtils() {
    }

    @Nullable
    public static ResourcesResponse getJsonFromApi(@Nullable Context context) {
        if (context == null) return null;

        String json;
        ResourcesResponse data = null;
        try {
            InputStream inputStream = context.getAssets().open("crowdin_api.json");
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, "UTF-8");
            Gson gson = new Gson();
            return gson.fromJson(json, ResourcesResponse.class);

        } catch (IOException e) {
            return data;
        }
    }

}

File: crowdin/src/main/java/com/crowdin/platform/utils/FileUtils.kt
package com.crowdin.platform.utils

import android.content.Context
import com.crowdin.platform.api.ResourcesResponse
import com.google.gson.Gson

internal object FileUtils {

    fun getJsonFromApi(context: Context?): ResourcesResponse? {
        if (context == null) return null

        val json: String
        val data: ResourcesResponse? = null
        return try {
            val inputStream = context.assets.open("crowdin_api.json")
            val size = inputStream.available()
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer)

            Gson().fromJson(json, ResourcesResponse::class.java)

        } catch (e: Throwable) {
            data
        }
    }
}
